app/lib/searching.rb

```diff
@@ -33,21 +38,24 @@ def initialize(params)
   end
 
   def count
-    return 10 if params["count"].blank?
+    return 10 if params["count"].blank? || params["count"].to_i.negative?
+    return 100 if params["count"].to_i > 100 && [params["host-a"], params["host-b"]].include?("staging")
     return 1000 if params["count"].to_i > 1000
     params["count"]
   end
 
   def call
-    rummager = GdsApi::Rummager.new(Plek.new.find('rummager'))
-    findings_new_left = rummager.search(
+    rummager_a = GdsApi::Rummager.new(HOSTS[params["host-a"]])
+    rummager_b = GdsApi::Rummager.new(HOSTS[params["host-b"]])
```

Comment:
what happens if the user selects an invalid host? could we have some sort or nice warning rather than an error.

Reply:
the user can't select an invalid host because they're in a drop down menu, unless there's a hacky way to do it?